docs/staging-use-cases.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Cluster API Reference Use Cases
+creation-date: 2019-04-16
+last-updated: 2019-04-16
+---
+
+# Cluster API Reference Use Cases
+
+This is a living document that serves as a reference and a staging area for use cases collected from the community during post-v1alpha1 project redesign.
+
+## Table of Contents
+
+* [Reference Use Cases](#cluster-api-reference-use-cases)
```

Comment:
For the ToC, we can use [doctoc](https://github.com/thlorenz/doctoc) to generate.